web/client/codechecker_client/credential_manager.py

```diff
@@ -98,22 +98,27 @@ def __init__(self):
         self.token_file = get_session_file()
         LOG.info("Checking for local valid sessions.")
 
-        if os.path.exists(self.token_file):
-            token_dict = load_json_or_empty(self.token_file, {},
-                                            "user authentication")
-            check_file_owner_rw(self.token_file)
-
-            self.__tokens = token_dict.get('tokens')
-            LOG.debug("Found session information for these hosts:")
-            for k, _ in self.__tokens.items():
-                LOG.debug("  %s", k)
-        else:
-            with open(self.token_file, 'w',
-                      encoding="utf-8", errors="ignore") as f:
+        with open(self.token_file, 'a+',
+                  encoding="utf-8", errors="ignore") as f:
+            portalocker.lock(f, portalocker.LOCK_EX)
+            f.seek(0)
+
+            try:
+                token_dict = json.loads(f.read())
+
+                check_file_owner_rw(self.token_file)
```

Comment:
Because it will print a warning message: https://github.com/Ericsson/codechecker/blob/81b2814803983ffd04da4652514f3c04c65baff6/web/codechecker_web/shared/env.py#L64-L68

Reply:
If the function purpose is that check the credential file and log the result of that check then "check_credential_file_access_and_warn" or similar would be better in this case.

Reply:
This is a function to check permission of any file not only credential files. So it is better to keep the function name general.

Also I don't think we should use the `and_warn` postfix. We have a lot of functions which can log warning/error messages. Do we need to use this postfix in all cases? I don't recommend it.